### PR TITLE
Prepare for v1.5.0 release

### LINKED ISSUE #60: Prepare the v1.5.0 release
Running list of changes before v1.5..0 release:

* Update all images in `README.md` to display the new validator options
* Update container README examples to use gcr released containers
* Update usage example to include `--depth` and `--version`
* Add gif at the top of the README to show usage
* Fix [depth bug](https://github.com/Boeing/config-file-validator/issues/69)
* Update README.md to specify that we only support Apple XML for PList

### LINKED ISSUE #69: --depth parameter showing inconsistent behaviour for different input paths
I was checking the depth feature and found the following issues

- The depth skip check is based on the path rather than the relative depth from the rootPath to the path.
```
$ go run cmd/validator/validator.go --depth=1 ./test/fixtures/subdir/
Summary: 0 succeeded, 0 failed
$ go run cmd/validator/validator.go --depth=2 ./test/fixtures/subdir/
Summary: 0 succeeded, 0 failed
$ go run cmd/validator/validator.go --depth=3 ./test/fixtures/subdir/
Summary: 0 succeeded, 0 failed
$ go run cmd/validator/validator.go --depth=4 ./test/fixtures/subdir/
    × test/fixtures/subdir/bad.json
        error: Error at line 3 column 14: invalid character ':' after array element
    × test/fixtures/subdir/bad.toml
        error: Error at line 6 column 6: toml: expected character =
    × test/fixtures/subdir/bad.yml
        error: yaml: line 3: did not find expected '-' indicator
    ✓ test/fixtures/subdir/good.json
Summary: 1 succeeded, 3 failed
```

- The depth check is based on counting separators in the path that can cause inconsistency if not checked from a clean path.
```
$ go run cmd/validator/validator.go --depth=3 test/fixtures/subdir/
    × test/fixtures/subdir/bad.json
        error: Error at line 3 column 14: invalid character ':' after array element
    × test/fixtures/subdir/bad.toml
        error: Error at line 6 column 6: toml: expected character =
    × test/fixtures/subdir/bad.yml
        error: yaml: line 3: did not find expected '-' indicator
    ✓ test/fixtures/subdir/good.json
Summary: 1 succeeded, 3 failed
exit status 1
$ go run cmd/validator/validator.go --depth=3 ./test/fixtures/subdir/
Summary: 0 succeeded, 0 failed
```

- The depth param set to 1 still parses files 2 directories down
```
$ go run cmd/validator/validator.go --depth=1 .
    ✓ .github/workflows/go.yml
    ✓ .github/workflows/release.yml
    ✓ test/fixtures/good.hcl
    ✓ test/fixtures/good.ini
    ✓ test/fixtures/good.json
    ✓ test/fixtures/good.properties
    ✓ test/fixtures/good.toml
    ✓ test/fixtures/good.yaml
```

This is the code in question:
https://github.com/Boeing/config-file-validator/blob/main/pkg/finder/fsfinder.go#L91
```go
err := filepath.WalkDir(fsf.PathRoot,
func(path string, dirEntry fs.DirEntry, err error) error {
	// determine if directory is in the excludeDirs list
	if dirEntry.IsDir() && fsf.Depth != nil && strings.Count(path, string(os.PathSeparator)) > *fsf.Depth {
		// Skip processing the directory
		return fs.SkipDir // This is not reported as an error by filepath.WalkDir
	}
```

### PR DESCRIPTION
**Pull request by kehoecj**
Cleanup and improvements before the v1.5.0 release

Closes #60 
Closes #69 

README has been cleaned up with a live demo: https://github.com/Boeing/config-file-validator/blob/23aaab2b0c6cc578248b128241a6da5c25d72be1/README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -98,14 +136,23 @@ validator --exclude-dirs=/path/to/search/tests /path/to/search
...
+Exclude file types in the search path. Available file types are `ini`, `json`, `yaml`, `yml`, `toml`, `xml`, `properties`, `hcl`, `csv` and `plist`
 
 ```
 validator --exclude-file-types=json /path/to/search
 ```
 
 ![Exclude File Types Run](./img/exclude_file_types.png)
 
+#### Customize recursion depth
+By default there is no recursion limt. If desired, the recursion depth can be set to an integer value. If depth is set to `0` recursion will be disabled and only the files in the search path will be validated.
```
Comment (jd4235) [inline README.md:148]:
```suggestion
By default there is no recursion limit. If desired, the recursion depth can be set to an integer value. If depth is set to `0` recursion will be disabled and only the files in the search path will be validated. 
```

File: README.md [hunk 2]
```diff
@@ -118,13 +165,32 @@ validator --reporter=json /path/to/search
...
-docker run -it --rm -v /path/to/config/files:/test config-file-validator:1.4.0 /test
+docker run -it --rm -v /path/to/config/files:/test config-file-validator:1.5.0 /test
 ```
 
 ![Docker Standard Run](./img/docker_run.png)
 
-## Building from source
-The project can be downloaded and built from source using an environment with golang 1.17+ installed. After successful build, the statically-linked binary can be moved to a location on your operating system PATH.
+## Build
+The project can be downloaded and built from source using an environment with golang 1.21 installed. After successful build, the binary can be moved to a location on your operating system PATH.
```
Comment (jd4235) [inline README.md:174]:
```suggestion
The project can be downloaded and built from source using an environment with golang 1.21 installed. After a successful build, the binary can be moved to a location on your operating system PATH.
```

File: index.md [hunk 3]
```diff
@@ -36,53 +38,89 @@
...
 * JSON
 * YAML
 * TOML
 * INI
-* Properties
 * HCL
-* PLIST
 * CSV
+* Properties
+* Apple PList XML
```
Comment (jd4235) [inline index.md:50]: Consider alphabetizing the list in all instances of this list.

File: index.md [hunk 4]
```diff
@@ -112,6 +158,15 @@ validator --exclude-file-types=json /path/to/search
 
 ![Exclude File Types Run](./img/exclude_file_types.png)
 
+#### Customize recursion depth
+By default there is no recursion limt. If desired, the recursion depth can be set to an integer value. If depth is set to `0` recursion will be disabled and only the files in the search path will be validated.
```
Comment (jd4235) [inline index.md:162]:
```suggestion
By default there is no recursion limit. If desired, the recursion depth can be set to an integer value. If depth is set to `0` recursion will be disabled and only the files in the search path will be validated. 
```

File: index.md [hunk 5]
```diff
@@ -124,13 +179,32 @@ validator --reporter=json /path/to/search
...
-docker run -it --rm -v /path/to/config/files:/test config-file-validator:1.4.0 /test
+docker run -it --rm -v /path/to/config/files:/test config-file-validator:1.5.0 /test
 ```
 
 ![Docker Standard Run](./img/docker_run.png)
 
-## Building from source
-The project can be downloaded and built from source using an environment with golang 1.17+ installed. After successful build, the statically-linked binary can be moved to a location on your operating system PATH.
+## Build
+The project can be downloaded and built from source using an environment with golang 1.21 installed. After successful build, the binary can be moved to a location on your operating system PATH.
```
Comment (jd4235) [inline index.md:188]:
```suggestion
The project can be downloaded and built from source using an environment with golang 1.21 installed. After a successful build, the binary can be moved to a location on your operating system PATH.
```

**Review by jd4235 — APPROVED** [review]
Nice touch on the GIF!

**Review by jackswiney — APPROVED** [review]

File: index.md (hunk 3 above)
Comment (kehoecj) [reply index.md:50]: Great idea! Will do

**Pull request merged**